### ISSUE TITLE
Develop Machine Learning Model

### ISSUE THREAD

**Issue by DanielJDufour** [GSA/time-estimator#3]
**Is your feature request related to a problem? Please describe.**
We want to be able to estimate how long it took to create some given source code based on the lines of code and the syntax used.  There's an assumption that certain code takes longer to write than others.  For example, Markdown is easier to write than JavaScript `reduce` functions.

**Describe the solution you'd like**
We want a function that takes in a path or url to a repository and calculates how long it took to produce that repository.

**Describe alternatives you've considered**
SLOC count is interesting.  We want to build on top of that and add another layer to it, which is the type of words used.

**Additional context**
If complete, this would be used as a factor in evaluating the person-hours of a source code repository and its value (in the sense of how much time is saved by reusing rather than reinventing the wheel).

**Comment by DanielJDufour** [comment]
Maybe try Stochastic Gradient Descent, which is supposed to be good with sparse data http://scikit-learn.org/stable/modules/sgd.html

**Comment by DanielJDufour** [comment]
We should create counts for term for each file extension, so we can programmatically create the vocabulary.
